### ISSUE TITLE
Cloud SQL (mysql and postgres) modules timeouts on user creation with replicas

### ISSUE THREAD

**Issue by romankaminski** [terraform-google-modules/terraform-google-sql-db#264]
### TL;DR

Creating postgres database with replica with v8.0.0 results in timeout creating sql user ( on "google_sql_user"). 
There is default timeout for google_sql_user = 10minutes. Cloning your module and expanding the timeout to 20m solved the problem.

### Expected behavior

"google_sql_user" should have timeouts parametrized, so we can expand the time needed for the creation.

### Observed behavior

timeout creating user for replica, what is failing the whole IaC

### Terraform Configuration

```hcl
testdata

project_id        = "******************"
instance_name     = "pgsql-testcase"
region            = "us-central1"
zone              = "us-central1-c"
availability_type = "REGIONAL"
user_labels = {
  environment_label = "pgsql_el_test"
  atosmanaged       = "true"
}
db_version          = "POSTGRES_13"
deletion_protection = "false"
backup_configuration = {
  enabled                        = "true"
  start_time                     = "3:30"
  location                       = "us-central1"
  point_in_time_recovery_enabled = "true"
  transaction_log_retention_days = "7"
  retained_backups               = null
  retention_unit                 = null
}
ip_configuration = {
  authorized_networks = []
  ipv4_enabled        = "false"
  require_ssl         = "true"
  private_network     = "projects/prj-d-shared-base-430c/global/networks/vpc-d-shared-base"
}
read_replicas = [{
    name = "-0"
    zone = "us-west1-a"
    tier = "db-f1-micro"
    disk_type = "PD_HDD"
    disk_autoresize = "false"
    disk_size = "10"
    user_labels = { environment = "test" }
    database_flags = [{
            name = "log_temp_files", value = 0
        },
        {
            name = "log_checkpoints", value = "on"
        },
        {
            name = "log_connections", value = "on"
        },
        {
            name = "log_disconnections", value = "on"
        },
       {
            name = "log_min_duration_statement", value = "-1"
        }]
    ip_configuration = {
        authorized_networks = []
        ipv4_enabled    = "false"
        private_network = "projects/************/global/networks/vpc-d-shared-base"
        require_ssl     = "true"
    }
    encryption_key_name = ""
}]
```

### Terraform Version

```sh
Terraform v1.0.10
on linux_amd64
```

### Additional information

It seems the problem didn't appear on v7.1.0 we had before and I see "provider" has been added to postgres database, so could this be effect of the google-beta provider?

**Comment by morgante** [comment]
Can you please share your provider configuration as well?

**Comment by romankaminski** [comment]
Here it is:

`terraform {
  required_version = ">=0.13"

  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "<4.0,>= 2.12"
    }
  }

}`

**Comment by morgante** [comment]
Thanks. I'm not sure why it is timing out. It shouldn't take longer than 10 minutes to create a new user.

Can you please open a support case with Google to debug this further? I would guess something non-Terraform is going on.

**Comment by romankaminski** [comment]
I might try that, but here is more details:

user creation and replica "alternates":
```
TestCloudSqlPgSQL 2021-12-15T09:41:18+01:00 logger.go:66: module.postgresql-db.google_sql_user.default[0]: Still creating... [9m20s elapsed]
TestCloudSqlPgSQL 2021-12-15T09:41:28+01:00 logger.go:66: module.postgresql-db.google_sql_database_instance.replicas["pgsql-testcase-replica-0"]: Still creating... [9m30s elapsed]
TestCloudSqlPgSQL 2021-12-15T09:41:28+01:00 logger.go:66: module.postgresql-db.google_sql_user.default[0]: Still creating... [9m30s elapsed]
TestCloudSqlPgSQL 2021-12-15T09:41:38+01:00 logger.go:66: module.postgresql-db.google_sql_database_instance.replicas["pgsql-testcase-replica-0"]: Still creating... [9m40s elapsed]
TestCloudSqlPgSQL 2021-12-15T09:41:38+01:00 logger.go:66: module.postgresql-db.google_sql_user.default[0]: Still creating... [9m40s elapsed]
TestCloudSqlPgSQL 2021-12-15T09:41:48+01:00 logger.go:66: module.postgresql-db.google_sql_database_instance.replicas["pgsql-testcase-replica-0"]: Still creating... [9m50s elapsed]
TestCloudSqlPgSQL 2021-12-15T09:41:48+01:00 logger.go:66: module.postgresql-db.google_sql_user.default[0]: Still creating... [9m50s elapsed]
TestCloudSqlPgSQL 2021-12-15T09:41:58+01:00 logger.go:66: module.postgresql-db.google_sql_database_instance.replicas["pgsql-testcase-replica-0"]: Still creating... [10m0s elapsed]
```

And the fail of user creation is:

```
TestCloudSqlPgSQL 2021-12-15T09:47:37+01:00 logger.go:66: ╷
TestCloudSqlPgSQL 2021-12-15T09:47:37+01:00 logger.go:66: │ Error: Error, failed to insert user default into instance pgsql-testcase-a8055e3e: googleapi: Error 409: Operation failed because another operation was already in progress., operationInProgress
TestCloudSqlPgSQL 2021-12-15T09:47:37+01:00 logger.go:66: │
```

Doesn't that implies something on the "terraform" level?

**Comment by morgante** [comment]
No. That error is being bubbled up from the API itself. Terraform is merely propagating it. You can tell from the "googleapi" prefix.
